loopy/target/c/__init__.py

```diff
@@ -899,14 +899,12 @@ def get_temporary_decls(self, codegen_state, schedule_index):
 
         ecm = self.get_expression_to_code_mapper(codegen_state)
 
-        for tv in sorted(
-                kernel.temporary_variables.values(),
-                key=lambda key_tv: key_tv.name):
+        for tv_name in sorted(sub_knl_temps):
+            tv = kernel.temporary_variables[tv_name]
             if not tv.base_storage:
                 # global temp vars are mapped to arguments or global
                 # declarations, no need to declare locally.
-                if tv.address_space != AddressSpace.GLOBAL and (
-                        tv.name in sub_knl_temps):
+                if tv.address_space != AddressSpace.GLOBAL:
```

Comment:
Not loving this change, as it just dumps all temporaries in all subkernels no matter what, whether they're used or not. `sub_knl_temps` would be unused/pointless after this. Closing in favor of #741 (or a version of it).

Reply:
The **bounds** of the loop had changed, but #741 is better as it fixes some more typos. I'm fine with closing this one.

Reply:
Agreed. I had missed that the iterable of the `for` loop had changed. Sorry!

Reply:
No worries at all! :)